### PR TITLE
Alzheimers classification using Swin Transfomer (46983156)

### PR DESCRIPTION
**Pull request by harri37**
# Overview
This pull request implements a Swin Transformer model for the classification of MRI images of the brain to detect Alzheimer's Disease (AD) versus no condition (NC).

## Introduction
Alzheimer's Disease (AD) is a neurodegenerative disorder characterized by β-amyloid (Aβ)­ containing extracellular plaques and tau-containing intracellular neurofibrillary tangles. The presentation of AD with short-term memory difficulty is most common, but impairment in expressive speech, visuospatial processing, and executive (mental agility) functions also occurs [[1]](#ref1).

Early detection of AD is crucial for the treatment and management of the disease. This project aims to classify MRI images as either AD or NC using the Swin Transformer architecture.

## Data
- **Dataset**: MRI images of the brain labeled as either Alzheimer's Disease (AD) or No Condition (NC).

## Model Architecture
The model leverages the Swin Transformer, which provides hierarchical feature extraction and attention mechanisms to capture spatial dependencies in the MRI images. This architecture allows for improved performance in image classification tasks.

## Files
- `models.py`: Implementation of the Swin Transformer model.
- `predict.py`: Functions to evaluate the model and make predictions
- `train.py`: Contains training process and handles data loading
- `dataset.py`: Functions for loading and preprocessing the MRI dataset.

## Dependencies
- torch
- matplotlib
- numpy
- scikit-learn
- tqdm
- scikit-learn
- seaborn
- pyyaml
- albumentations

### CODE REVIEW

**Comment by shaivikaaaa** [comment]
@gayanku

**Comment by aniketgupta17** [comment]
Observational Feedback 

File Organizing: Well-organized files.

Pull Request :
Correctly generated to Topic Recognition Branch .
The Description talks about the File Structure Clearly,  add about achieved Results .

Problem Solving:

The algorithm solves the problem appropriately . 
Accuracy in testing dataset: 0.82 

Model and functions:

It correctly uses PyTorch to construct the models and functions.
Code design: Good.

Code comment and docstring:
 
Good  Code Commenting . 
Missing function docstrings in some files . 
Difficulty: Hard.

Additional Comments:

Meaningful commits but could show progressive Logical Development
Good ReadMe design. However, It would be better if a conclusion and future improvements  appeared in the report.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>Header blocks. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>Good solution to problem. </td><td></td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting missing. Missing doc string.</td><td>-1</td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Hard. ViT (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). On time submission.</td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-1</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by shakes76** [comment]
Meaningful commit messages -1
